internal/ingress/convert_test.go

```diff
@@ -32,14 +34,23 @@ var v1TestIngress = &networkingv1.Ingress{
 		Name:      "test-networkingv1-ingress",
 		Namespace: "test-networkingv1-namespace",
 		Annotations: map[string]string{
-			"test.key": "test.value",
+			"test.key":                    "test.value",
+			"kubernetes.io/ingress.class": "bogus-ingress-class",
 		},
 		Labels: map[string]string{
 			"labelkey": "labelvalue",
 		},
 	},
 	Spec: networkingv1.IngressSpec{
-		IngressClassName: pointer.String("bogus-ingress-class"),
+		// As discussed in https://github.com/jetstack/cert-manager/issues/4537
+		// the IngressClassName field is not directly equivalent to the
+		// kubernetes.io/ingress.class annotation. According to the Ingress graduation KEP:
+		// https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/758-ingress-api-group/README.md
+		// the annotation, if present, overrides this field.
+		//
+		// As the goal of cert-manager is to be as widely compatible as possible, we can only use the
+		// annotation to describe an ingress class.
```

Comment:
I was thinking: if the need arises (which it won't IMHO), we could add a supplementary field `ingressClass` or something like that on the Issuer.